linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -1247,7 +1351,8 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
      *  A side-effect free expression can be elided if its result is not used.
      */
```

Comment:
Maybe it is worth expanding the comment to explain that for longs, this requires they are splittable (unless explicitly allowed).

FWIW: The presence of `allowUnsplittableLongs` in the param list makes this less necessary, but it might be helpful anyways.